l1-contracts/src/core/Rollup.sol

```diff
@@ -1049,23 +1068,28 @@ contract Rollup is EIP712("Aztec Rollup", "1"), Leonidas, IRollup, ITestRollup {
   }
 
   /**
-   * @notice  Validate an L2 block's blob. TODO: edit for multiple blobs per block
+   * @notice  Validate a blob.
    * Input bytes:
    * input[:32]     - versioned_hash
    * input[32:64]   - z
    * input[64:96]   - y
    * input[96:144]  - commitment C
    * input[144:192] - proof (a commitment to the quotient polynomial q(X))
    *  - This can be relaxed to happen at the time of `submitProof` instead
-   *
+   * @notice Apparently there is no guarantee that the blobs will be processed in the order sent
```

Comment:
So a few things here - one, I got a lot of `stack too deep` errors and so separated out a lot of functionality, this included. With other reworks, this now might work as one function, but also might not.
Secondly, this got very complex to manage with `checkBlob`, which allows us to throw errors correctly and actually simulate txs with viem without it falling over, because it doesn't support blobs very well. So this overall isn't a final version, ideally we remove `checkBlob` and squash into one fn. I can attempt this now but given deadlines I thought it more of a future thing.
Finally, I also don't know whether our use of 4844 txs does or does not guarantee ordering of blobs. I would prefer to confirm that 100% rather than add a confusing loop then having to reassign blob hashes and public inputs coming out of it! But yes, we would need to match them up as above if the order isn't guaranteed. All the things I ran had the blobs in the order I sent.

Reply:
Yep, that all makes sense! 
Please do open issues in the viem repo, if you have improvements in mind :)